### PR TITLE
Inject private git webserver in mlflow

### PR DESCRIPTION
**Pull request by amrqt**
## What changes are proposed in this pull request?

Allow usage of a private version control system when displaying experiment source and version links

We want to package the app without having to rebuild the UI with custom VCS settings so these settings are passed at runtime to the Python backend through environment variables.

In `mlflow/server/js/src/utils/Utils.js`:
* Added cases to `Utils.getGitRepoUrl()` and `Utils.getGitCommitUrl()` to generate urls to a private vcs. 
* Added `getPrivateVcsRegex` and `getPrivateVcsUrl` which are called when Utils.js is loaded to set the values we need for regex matching and url generation.

This implementation is currently not ideal as it produces three synchronous calls to the backend when Utils.js is loaded and allows only limited pattern matching for url generation. A better way would be to have async calls to the backend push back values to view components.

## How is this patch tested?

Added tests for the new JS functions, and manual testing.
 
### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Displays links to a private version control system.

### What component(s) does this PR affect?

- [X] UI
- [ ] CLI
- [ ] API
- [x] REST-API
- [ ] Examples
- [x] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/search-syntax.rst [hunk 1]
```diff
@@ -89,7 +89,6 @@ You can search using two run attributes contained in :py:class:`mlflow.entities.
```
Comment (jdlesage) [inline docs/source/search-syntax.rst]: This change is expected to be in this PR ?

File: mlflow/server/__init__.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 from mlflow.server.handlers import get_artifact_handler, STATIC_PREFIX_ENV_VAR, _add_static_prefix
 from mlflow.utils.process import exec_cmd
 
+
```
Comment (jdlesage) [inline mlflow/server/__init__.py]: Problably a mistake when you staged your changes...

File: mlflow/server/js/package.json [hunk 3]
```diff
@@ -47,7 +48,7 @@
     "yup": "^0.26.3"
   },
   "devDependencies": {
-    "babel-eslint": "8.2.2",
+    "babel-eslint": "^7.2.3",
```
Comment (jdlesage) [inline mlflow/server/js/package.json]: Why did you downgrade the version ?

File: mlflow/server/js/package.json [hunk 4]
```diff
@@ -63,8 +64,8 @@
     "eslint-plugin-react": "7.4.0",
     "eslint-plugin-standard": "3.0.1",
     "jest-localstorage-mock": "^2.3.0",
-    "react-app-rewired": "^1.5.2",
-    "react-app-rewire-define-plugin": "1.0.0"
+    "react-app-rewire-define-plugin": "1.0.0",
+    "react-app-rewired": "^1.5.2"
   },
```
Comment (jdlesage) [inline mlflow/server/js/package.json]: Could you keep original order, please ?

File: mlflow/server/js/src/utils/Utils.js [hunk 5]
```diff
@@ -445,6 +447,61 @@ class Utils {
...
+      window.privateVcsRegex = Utils.getPrivateVcsRegex();
+    }
+    if (!window.privateVcsRepo) {
+      window.privateVcsRepo = Utils.getPrivateVcsUrl('repo');
+    }
+    if (!window.privateVcsCommit) {
+      window.privateVcsCommit = Utils.getPrivateVcsUrl('commit');
+    }
+
+    if (window.privateVcsRegex && window.privateVcsRepo && window.privateVcsCommit) {
```
Comment (jdlesage) [inline mlflow/server/js/src/utils/Utils.js]: return express

File: mlflow/server/__init__.py (hunk 2 above)
Comment (amrqt) [reply mlflow/server/__init__.py]: done

File: mlflow/server/js/package.json (hunk 3 above)
Comment (amrqt) [reply mlflow/server/js/package.json]: done

File: mlflow/server/js/package.json (hunk 4 above)
Comment (amrqt) [reply mlflow/server/js/package.json]: done

**Pull request merged**